src/stt/include/stt/flute.h

```diff
@@ -56,116 +55,166 @@ using stt::Tree;
 #define FLUTE_POSTFILE "POST9.dat"  // LUT for POST (Steiner Tree)
 #define FLUTE_D 9  // LUT is used for d <= FLUTE_D, FLUTE_D <= 9
 
-// User-Callable Functions
-// Delete LUT tables for exit so they are not leaked.
-void deleteLUT();
-int flute_wl(int d,
-             const std::vector<int>& x,
-             const std::vector<int>& y,
-             int acc);
-Tree flute(const std::vector<int>& x, const std::vector<int>& y, int acc);
-int wirelength(Tree t);
-void plottree(Tree t);
-void write_svg(Tree t, const char* filename);
-
-// Other useful functions
-int flutes_wl_LD(int d,
+#if FLUTE_D <= 7
+#define MGROUP 5040 / 4  // Max. # of groups, 7! = 5040
+#define MPOWV 15         // Max. # of POWVs per group
+#elif FLUTE_D == 8
+#define MGROUP 40320 / 4  // Max. # of groups, 8! = 40320
+#define MPOWV 33          // Max. # of POWVs per group
+#elif FLUTE_D == 9
+#define MGROUP 362880 / 4  // Max. # of groups, 9! = 362880
+#define MPOWV 79           // Max. # of POWVs per group
+#endif
+
+class Flute
+{
+ public:
+  struct csoln
+  {
+    unsigned char parent;
+    unsigned char seg[11];  // Add: 0..i, Sub: j..10; seg[i+1]=seg[j-1]=0
+    unsigned char rowcol[FLUTE_D - 2];  // row = rowcol[]/16, col = rowcol[]%16,
+    unsigned char neighbor[2 * FLUTE_D - 2];
+  };
+  using LUT_TYPE = struct csoln***;
+  using NUMSOLN_TYPE = int**;
+
+ public:
+  Flute() = default;
+  ~Flute() { deleteLUT(); }
+
+  // User-Callable Functions
+  // Delete LUT tables for exit so they are not leaked.
+  void readLUT();
+  void makeLUT(LUT_TYPE& LUT, NUMSOLN_TYPE& numsoln);
+  void initLUT(int to_d, LUT_TYPE LUT, NUMSOLN_TYPE numsoln);
+  void ensureLUT(int d);
+  void deleteLUT();
+  void deleteLUT(LUT_TYPE& LUT, NUMSOLN_TYPE& numsoln);
```

Comment:
It looks like only ensureLUT needs to be public.

Reply:
Only the methods flute, plottree, write_svg, and flutes need to be public